src/tezos/TezosMessageCodec.ts

```diff
@@ -0,0 +1,242 @@
+    if (getOperationType(hexOperationType) !== "transaction") {
+      throw new Error("Provided operation is not a transaction.");
+    }
+
+    let fieldoffset = 0;
+    let branch = TezosMessageUtils.readBranch(transactionMessage.substring(fieldoffset, fieldoffset + 64));
+
+    fieldoffset += 64 + 2; // branch + type
+    let source = "";
+    if (transactionMessage.substring(fieldoffset, fieldoffset + 4) === "0000") {
+      fieldoffset += 4;
+      source = TezosMessageUtils.readAddress(transactionMessage.substring(fieldoffset, fieldoffset + 40));
+    } else {
+      fieldoffset += 2;
+      source = TezosMessageUtils.readAddress(transactionMessage.substring(fieldoffset, fieldoffset + 40), 'kt1');
+      fieldoffset += 2;
+    }
+
+    fieldoffset += 40;
+    let feeInfo = TezosMessageUtils.findInt(transactionMessage, fieldoffset);
+
+    fieldoffset += feeInfo.length;
+    let counterInfo = TezosMessageUtils.findInt(transactionMessage, fieldoffset);
+
+    fieldoffset += counterInfo.length;
+    let gasInfo = TezosMessageUtils.findInt(transactionMessage, fieldoffset);
+
+    fieldoffset += gasInfo.length;
+    let storageInfo = TezosMessageUtils.findInt(transactionMessage, fieldoffset);
+
+    fieldoffset += storageInfo.length;
+    let amountInfo = TezosMessageUtils.findInt(transactionMessage, fieldoffset);
+
+    fieldoffset += amountInfo.length;
+    let target = "";
+    if (transactionMessage.substring(fieldoffset, fieldoffset + 4) === "0000") {
+      fieldoffset += 4;
+      target = TezosMessageUtils.readAddress(transactionMessage.substring(fieldoffset, fieldoffset + 40));
+      fieldoffset += 40 + 2;
+    } else if (transactionMessage.substring(fieldoffset, fieldoffset + 2) === "01"){
+      fieldoffset += 2;
+      target = TezosMessageUtils.readAddress(transactionMessage.substring(fieldoffset, fieldoffset + 40), 'kt1');
+      fieldoffset += 40 + 4;
+    }
+
+    let next;
+    if (transactionMessage.length > fieldoffset) {
+      next = getOperationType(transactionMessage.substring(fieldoffset, fieldoffset + 2));
+    }
+
+    const transaction: Operation = {
+      kind: "transaction",
+      source: source,
+      destination: target,
+      amount: amountInfo.value + "",
+      fee: feeInfo.value + "",
+      gas_limit: gasInfo.value + "",
+      storage_limit: storageInfo.value + "",
+      counter: counterInfo.value + ""
+    };
+
+    const envelope: OperationEnvelope = {
+      operation: transaction,
+      branch: branch,
+      next: next,
+      nextoffset: next ? fieldoffset : -1
+    }
+
+    return envelope;
+  }
+
+  /**
+   * Parse a reveal message possibly containing siblings.
+   * @param {string} revealMessage Encoded reveal-type message
+   * @param {boolean} isFirst Flag to indicate first operation of Operation Group.
+   */
+  export function parseReveal(revealMessage, isFirst = true): OperationEnvelope {
```

Comment:
What are the types of these parameters?

Reply:
`export function parseReveal(revealMessage: string, isFirst:boolean = true)`